orc8r/protos/bootstrapper.proto

```diff
@@ -71,3 +73,83 @@ service Bootstrapper {
   // Returns signed certificate.
   rpc RequestSign (Response) returns (Certificate) {}
 }
+
+// --------------------------------------------------------------------------
+// Protos for CloudRegistration and Registration servicer
+// --------------------------------------------------------------------------
+service CloudRegistration {
+  // GetToken saves and sends out a token related to input:networkID and input:logicalID
+  // It will refresh the token if input:refresh is true or token is expired
+  rpc GetToken(GetTokenRequest) returns (GetTokenResponse) {}
+
+  // GetInfoForGatewayRegistration sends out information needed for gateways to register themselves
+  rpc GetGatewayRegistrationInfo(GetGatewayRegistrationInfoRequest) returns (GetGatewayRegistrationInfoResponse) {}
+
+  // GetGatewayPreregisterInfo sends out information that has been set when the operator preregistered
+  // the device with networkID and logicalID
+  rpc GetGatewayDeviceInfo(GetGatewayDeviceInfoRequest) returns (GetGatewayDeviceInfoResponse) {}
+}
+
+service Registration {
+  // Register should be called by the gateway device
+  rpc Register(RegisterRequest) returns (RegisterResponse) {}
+}
+
+message GetTokenRequest {
+  GatewayDeviceInfo gateway_device_info = 1;
+  // refresh is true if a new token should be generated regardless of old token timeout
+  bool refresh = 2;
+}
+
+message GetTokenResponse {
+  // token is a nonce prepended by bootstrapper.tokenPrepend
+  string token = 1;
+  google.protobuf.Timestamp timeout = 2;
+}
+
+message GetGatewayRegistrationInfoRequest {
+}
+
+message GetGatewayRegistrationInfoResponse {
+  string root_ca = 1;
+  string domain_name = 2;
+}
+
+message GetGatewayDeviceInfoRequest {
+  // token is a nonce prepended by bootstrapper.tokenPrepend
+  string token = 1;
+}
+
+message GetGatewayDeviceInfoResponse {
+  oneof response {
+    GatewayDeviceInfo gateway_device_info = 1;
+    string error = 2;
+  }
+}
+
+message RegisterRequest {
+  // token is a nonce prepended by bootstrapper.tokenPrepend
+  string token = 1;
+  AccessGatewayID hwid = 2;
+  // challenge_key is gateway's long-term public key
+  ChallengeKey challenge_key = 3;
+}
+
+message RegisterResponse {
+  oneof response {
+    // control_proxy is the OK that the gateway has registered properly
+    string control_proxy = 1;
```

Comment:
What sort of values would `control_proxy` taken on if the gateway has registered correctly?

Reply:
added example in protos